### DISCUSSION TITLE
Is it possible to generate this routes?

### DISCUSSION THREAD

**Discussion by MrAmericanMike** [sveltejs/kit#7807] [Q&A] [answered]
The problem is explained on the README.md of the template I'm going to link, but posting it here also for anyone that knows or can help.

Thanks in advance.

Link to StackBlitz Example: https://stackblitz.com/edit/sveltekit-dynamicstaticroutes?file=README.md

*****

The idea is simple. Having the routes that this project includes (See StackBlitz) I want to be able to generate a static site with a final folder structure looking like this.

```
build/
├── index.html
├── about/
│   └── index.html
└── post/
    ├── post-01/
    │   └── index.html
    ├── post-02/
    │   └── index.html
    └── post-03/
        └── index.html
```

As the project is set right now, the website itself works, however the `post/` folder doesn't exist on the final build and this route is handled dynamically. (Because of `src/routes/post/[slug]/+page.js` containing `export const prerender = false;` Which of course, changing this to true would cause an error)

#### Given that the 3 post routes are known at build time. Is there any way to have the static adapter generate them?

**Comment by david-plugge** [comment, marked as answer]
Try enabling ssr

**Reply by MrAmericanMike** [reply]
> Try enabling ssr

Thanks a lot, this did the trick. Wouldn't have think that enabling SSR was the way to go when I want an static website (Without server side code)

I changed on +layout.js `export const ssr = true;` to true and the +page.js of the [slug] folder `export const prerender = true;` to true also, and that generated the routes.

**Reply by geoffrich** [reply]
I think the issue was:

1. SvelteKit figures out which routes to prerender by crawling the generated HTML, i.e. looking for "/page/1", "/page/2", etc. links
2. Because the root `+page.svelte` was not SSR'd, the generated HTML did not have the "/page" links in it, so SvelteKit could not determine which "/page" links to generate
3. By SSRing the root layout, the generated HTML now has the "/page" links in it, so SvelteKit can successfully crawl them and generate those files

An alternate solution would be to set [prerender.entries](https://kit.svelte.dev/docs/configuration#prerender) to include the "/page" links, but turning on SSR is the better choice.
